### ISSUE TITLE
Support strict-prototypes

### ISSUE THREAD

**Issue by orlitzky** [flintlib/flint#1683]
Flint-3.0.1 triggers some strict prototype warnings that are easy to find by turning on `-Werror=strict-prototypes`. For example,

```
/var/lib/portage/tmp/portage/sci-mathematics/flint-3.0.1/work/flint-3.0.1/src/profiler.h:131:8: error: function declaration isn't a prototype [-Werror=strict-prototypes]
  131 | double get_cycle_counter()
      |        ^~~~~~~~~~~~~~~~~
/var/lib/portage/tmp/portage/sci-mathematics/flint-3.0.1/work/flint-3.0.1/src/profiler.h:157:6: error: function declaration isn't a prototype [-Werror=strict-prototypes]
  157 | void init_all_clocks()
      |      ^~~~~~~~~~~~~~~
/var/lib/portage/tmp/portage/sci-mathematics/flint-3.0.1/work/flint-3.0.1/src/profiler.h:190:6: error: function declaration isn't a prototype [-Werror=strict-prototypes]
  190 | void prof_start()
      |      ^~~~~~~~~~
/var/lib/portage/tmp/portage/sci-mathematics/flint-3.0.1/work/flint-3.0.1/src/profiler.h:196:6: error: function declaration isn't a prototype [-Werror=strict-prototypes]
  196 | void prof_stop()
      |      ^~~~~~~~~
```

In this case, it looks like that's the first declaration of those functions, and the empty parens indicate an unspecified number of arguments of unspecified type. That's been deprecated "forever" in C, and isn't a big deal right now, but the C23 standard plans to make it an error, so it would be nice to fix before people start reporting build failures with clang-19 or whatever compiler release is contemporaneous with the new standard.

**Comment by albinahlback** [comment]
This was fixed in 0a6d4100e7e1bb544917e36cfcce7e3dadec51fb. I will let @fredrik-johansson take care of any backports.

**Comment by orlitzky** [comment]
> This was fixed in [0a6d410](https://github.com/flintlib/flint/commit/0a6d4100e7e1bb544917e36cfcce7e3dadec51fb). I will let @fredrik-johansson take care of any backports.

Oh, cool, thanks. I don't think it's crucial to backport yet. Unless you get tired of people re-reporting it :)
